x-pack/legacy/plugins/apm/server/lib/metrics/transform_metrics_chart.ts

```diff
@@ -49,23 +50,23 @@ export function transformDataToMetricsChart<Params extends AggregatedParams>(
   chartBase: ChartBase
 ) {
   const { aggregations, hits } = result;
-  const { timeseriesData } = aggregations;
+  const timeseriesData = idx(aggregations, _ => _.timeseriesData);
 
   return {
     title: chartBase.title,
     key: chartBase.key,
     yUnit: chartBase.yUnit,
     totalHits: hits.total,
     series: Object.keys(chartBase.series).map((seriesKey, i) => {
-      const agg = aggregations[seriesKey];
+      const agg = aggregations ? aggregations[seriesKey] : { value: null };
```

Comment:
Can you do this instead?
```suggestion
      const overallValue = idx(aggregations, _=> _[seriesKey].value);
```

Reply:
Hmm, it's not the same though. It creates a union type of `{ value: number | null} | undefined` which leads to errors further down the line. We would have to change line 68 to:
```ts
overallValue: agg ? agg.value : null,
```

still worth it?

Reply:
Oh, sorry, misread your suggestion!

Reply:
Much better 👍

Reply:
I think you need to adjust `ByteFormatter` in formatters to also accept `undefined`. `asBytes` and `asDynamicBytes` and for consistency so should the other nyt formatting function. This was a previous oversight but a good opportunity to fix it I think.

Reply:
... aka so doing 
```js
overallValue: idx(aggregations, _=> _[seriesKey].value)
```

should still be possible.